packages/payment/src/loaders/providers.ts

```diff
@@ -1,24 +1,18 @@
 import { moduleProviderLoader } from "@medusajs/modules-sdk"
-
 import { LoaderOptions, ModuleProvider, ModulesSdkTypes } from "@medusajs/types"
-import { Lifetime, asFunction } from "awilix"
+import { Lifetime, asFunction, asValue } from "awilix"
 
 const registrationFn = async (klass, container, pluginOptions) => {
-  container.register({
-    [`payment_provider_${klass.prototype}`]: asFunction(
-      (cradle) => new klass(cradle, pluginOptions),
-      {
+  Object.entries(pluginOptions.config).map(([k, v]) => {
```

Comment:
todo: check that config is present, also, remove `k`, `v` and use instead `name`, `config`